CodePush.js

```diff
@@ -75,45 +78,30 @@ async function checkForUpdate(deploymentKey = null, handleBinaryVersionMismatchC
            * @type {string}
            */
           const runtimeVersion = sharedCodePushOptions.runtimeVersion;
-
-
-          // NOTE: It is not necessary to implement it in three steps. However, it is recommended to implement unit-test-friendly.
-          /**
-           * TODO 1: find latest release in releaseHistory
-           * @return {ReleaseInfo}
-           */
-          function findLatestRelease(releaseHistory) {}
-
-          /**
-           * TODO 2: check if the update is mandatory
-           * @return {boolean}
-           */
-          function checkIsMandatory(runtimeVersion, releaseHistory) {}
-
-          /**
-           * TODO 3: determine whether to rollback and execute it
-           * @return {boolean}
-           */
-          function shouldRollback(runtimeVersion, releaseHistory) {}
-          if (shouldRollback(runtimeVersion, releaseHistory)) {
-            // rollback
+          const versioning = new sharedCodePushOptions.versioning(releaseHistory);
```

Comment:
We should add type annotations for `versioning`.

Then, I found that there were several unnecessary arguments being passed.
![image](https://github.com/user-attachments/assets/17194bd3-b46f-47bb-8cb6-7551eb0fef76)

Reply:
@floydkim Oh, included in [fa234fc](https://github.com/Soomgo-Mobile/react-native-code-push/pull/27/commits/fa234fc4f448f8b46b9578e5248663f04fcce7c6)

Reply:
@changwoolab

It would be better to add a type annotation for `versioning` 😄 
Currently, it's inferred as `any`.

Reply:
@floydkim Included in [f78224b](https://github.com/Soomgo-Mobile/react-native-code-push/pull/27/commits/f78224b5df74c1af7d50f03cf19451389008e6d6) !